Source/Core/Core/NetPlayServer.cpp

```diff
@@ -2067,9 +2068,17 @@ bool NetPlayServer::SyncCodes()
   }
   // Sync Gecko Codes
   {
+    std::vector<Gecko::GeckoCode> s_codes = Gecko::LoadCodes(globalIni, localIni);
+
+#ifdef USE_RETRO_ACHIEVEMENTS
+    {
+      std::lock_guard lg{AchievementManager::GetInstance().GetLock()};
+      AchievementManager::GetInstance().FilterApprovedGeckoCodes(s_codes, game_id);
+    }
+#endif  // USE_RETRO_ACHIEVEMENTS
+
     // Create a Gecko Code Vector with just the active codes
-    std::vector<Gecko::GeckoCode> s_active_codes =
-        Gecko::SetAndReturnActiveCodes(Gecko::LoadCodes(globalIni, localIni));
+    std::vector<Gecko::GeckoCode> s_active_codes = Gecko::SetAndReturnActiveCodes(s_codes);
```

Comment:
I know this one isn't your fault, but let's clean it up while we're here, because it's very misleading.

```suggestion
    std::vector<Gecko::GeckoCode> active_codes = Gecko::SetAndReturnActiveCodes(s_codes);
```